test/http_redirect_test.dart

```diff
@@ -0,0 +1,472 @@
+// Copyright (c) 2018, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import "dart:async";
+
+import 'package:http_io/http_io.dart';
+import 'package:test/test.dart';
+
+Future<HttpServer> setupServer() {
+  Completer completer = new Completer();
```

Comment:
No need for a completer here, just return the server directly.

Reply:
Done